acme/api/account.go

```diff
@@ -205,3 +224,83 @@ func (h *Handler) GetOrdersByAccountID(w http.ResponseWriter, r *http.Request) {
 	api.JSON(w, orders)
 	logOrdersByAccount(w, orders)
 }
+
+// validateExternalAccountBinding validates the externalAccountBinding property in a call to new-account
+func (h *Handler) validateExternalAccountBinding(ctx context.Context, nar *NewAccountRequest) (*acme.ExternalAccountKey, error) {
+
+	prov, err := provisionerFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	acmeProv, ok := prov.(*provisioner.ACME) // TODO: rewrite into providing configuration via function on acme.Provisioner
+	if !ok || acmeProv == nil {
+		return nil, acme.NewErrorISE("provisioner in context is not an ACME provisioner")
+	}
+
+	shouldSkipAccountBindingValidation := !acmeProv.RequireEAB
```

Comment:
Do we need a variable for this or can we just check `if !acmeProv.RequireEAB`?

Reply:
More of a personal preference, so that it's clear what is happening. Removed it in this case; it's not really hard to understand what's happening without the var.